### PR TITLE
Feat/gh43-44-67-68 justification dialog transfer history; fixes #43 #44 #67 #68

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEH-CommonJ/pulls) open
- [x] I have verified that I am following the DEH-CommonJ [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEH-CommonJ/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

- [x] Implemented the justification dialog and local exchange history

### CODE REVIEW

File: src/main/java/HubController/HubController.java [hunk 1]
```diff
@@ -559,6 +593,57 @@ public void Write(ThingTransaction transaction) throws DalWriteException, Comple
         this.session.write(transaction.finalizeTransaction()).join();
     }    
 
+    /**
+     * Tries to create a {@linkplain LogEntry} base on the input from the {@linkplain LogEntryDialog}
+     * 
+     * @param transaction the {@linkplain ThingTransaction}
+     * @return a value indicating whether the whole transaction should be cancelled based on the dialog result
+     * @throws TransactionException
```
Comment (antoineatrhea) [inline src/main/java/HubController/HubController.java:601]: Missing doc

File: src/main/java/HubController/HubController.java [hunk 2]
```diff
@@ -559,6 +593,57 @@ public void Write(ThingTransaction transaction) throws DalWriteException, Comple
...
+        return true;
+    }
+    
+    /**
+     * Adds a new {@linkplain ModelLogEntry} record to the {@linkplain EngineeringModel}
+     *  and registers the change to the provided {@linkplain ThingTransaction}
+     * 
+     * @param content the {@linkplain String} content
+     * @param the {@linkplain ThingTransaction}
+     * @throws TransactionException
```
Comment (antoineatrhea) [inline src/main/java/HubController/HubController.java:623]: Missing doc

File: src/main/java/ViewModels/ExchangeHistory/ExchangeHistoryRenderDataProvider.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    public Icon getIcon(Object rowViewModel)
+    {
+        if(rowViewModel instanceof ExchangeHistoryTimeStampRowViewModel)
+        {
+            return ImageLoader.GetIcon();
+        }
+        
+        return null;
+    }
+}
```
Comment (antoineatrhea) [inline src/main/java/ViewModels/ExchangeHistory/ExchangeHistoryRenderDataProvider.java:124]: Missing EOF

File: src/main/java/Views/ExchangeHistory/package-info.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * The DEH-CommonJ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package Views.ExchangeHistory;
```
Comment (antoineatrhea) [inline src/main/java/Views/ExchangeHistory/package-info.java:24]: Missing EOF

File: src/main/java/Views/Interfaces/IDialog.java [hunk 5]
```diff
@@ -35,21 +35,21 @@
...
 
      /**
       * Gets the {@linkplain dialogResult}
       * 
-      * @return a {@linkplain Boolean}
+      * @return a {@linkplain #TResult}
       */
-     Boolean GetDialogResult();
+    TResult GetDialogResult();
 }
```
Comment (antoineatrhea) [inline src/main/java/Views/Interfaces/IDialog.java:55]: Missing EOF

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
minor changes

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**